launcher/launcher_test.go

```diff
@@ -291,18 +302,20 @@ func (s *LauncherTestSuite) TestWaitForUpdates() {
 			s.NoError(err)
 
 			go func(ic incomingConfigs) {
-				defer close(mocks.reloader.ch)
-
 				for _, c := range ic {
 					time.Sleep(c.after)
 					mocks.reloader.ch <- c.config
 				}
+			}(tcase.configs)
 
+			go func() {
 				time.Sleep(100 * time.Millisecond)
-			}(tcase.configs)
+				mocks.cancel()
+				close(mocks.reloader.ch)
+			}()
 
 			err = sut.run()
-			s.Error(err)
+			s.NoError(err)
```

Comment:
Why did you have to change it?

Reply:
I notice that it always returns an error after the channel is been closed (https://github.com/elastic/cloudbeat/pull/457/files#diff-29b1f6ad1656f858176a4986694417256c06f97b53ccaaad97c5add2a6b01ea1L294), so I have added [this](https://github.com/elastic/cloudbeat/pull/457/files#diff-29b1f6ad1656f858176a4986694417256c06f97b53ccaaad97c5add2a6b01ea1R311-R315) block to close it gracefully.